### DISCUSSION TITLE
Native using Blazor/Razor syntax

### DISCUSSION THREAD

**Discussion by DevronB** [dotnet/maui#1835] [Q&A] [answered]
Is it possible to use the Blazor/Razor syntax in maui to create a native app without requiring an underlying webview. Basically I want to compare using Xaml, Razor and Comet for native development. I'd like some of my code/components to be shared between Android and Wear OS but Wear OS does not support WebView. I thought this might be possible as Blazor Desktop is running on Net6 and not wasm.  It's also possible I'm missing some important aspects here... is Blazor Desktop/Razor really an option without webview?

**Comment by GeraudFabien** [comment]
Blazor use html/css you can use [Maui blazor binding](https://devblogs.microsoft.com/aspnet/unified-ui-mobile-blazor-bindings-preview-5/). But you always need webview to render html/css.

**Reply by DevronB** [reply]
But doesn't Blazor Desktop/MAUI Blazor Bindings work as a programming/binding model without Html/CSS? Can't I use XAML with blazor/razor syntax? I understand a Hybrid model would require WebView for the html/Blazor part, but if I'm not using Hybrid, just native and XAML can I use this?

From the Native example you linked to there is no HTML.
![image](https://user-images.githubusercontent.com/27395345/127602723-5dbf0c94-350e-4928-a24b-ec03bd355071.png)

**Reply by redradist** [reply]
@GeraudFabien 
> But you always need webview to render html/css.

It is not true !!

With Maui Blazor binding you use only Blazor syntax not WebView ...
WebView was added later after work on Maui blazor binding started

**Reply by GeraudFabien** [reply]
Yes you are totally correct. But  Maui blazor binding only add the syntax (it doesn't use html/CSS). Nothings more so my sentence is actually true. It depend on what you want.

**Comment by SarthakGz** [comment, marked as answer]
@DevronB  yes you can totally do that check this out https://github.com/dotnet/MobileBlazorBindings
It's experimental though and didn't get any recents update don't know if they are still investing time in it
